docs/content/rust-components.md

```diff
@@ -286,3 +285,41 @@ the `target/wasm32-wasi/release` directory and looking for `.wasm` files)
 - make sure the path and name of the Wasm module in `target/wasm32-wasi/release`
 match `source` field in the component configuration (the `source` field contains
 the path to the Wasm module, relative to `spin.toml`)
+
+## Manually creating new projects with Cargo
+
+The recommended way of creating new Spin projects is by starting from a template.
+This section shows how to  manually create a new project with Cargo.
+
+When creating a new Spin projects with Cargo, you should use the `--lib` flag.
+
+```console
+$ cargo init --lib
+```
+
+A `Cargo.toml` with standard Spin dependencies looks like this:
+
+```toml
+[package]
+name = "your-app"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = [ "cdylib" ]
+
+[dependencies]
+# Useful crate to handle errors.
+anyhow = "1"
+# Crate to simplify working with bytes.
+bytes = "1"
+# General-purpose crate with common HTTP types.
+http = "0.2"
+# The Spin SDK.
+spin-sdk = { git = "https://github.com/fermyon/spin" }
+# Crate that generates Rust Wasm bindings from a WebAssembly interface.
+wit-bindgen-rust = { git = "https://github.com/bytecodealliance/wit-bindgen", rev = "2f46ce4cc072107153da0cefe15bdc69aa5b84d0" }
+```
+
+At the time of this writing, `wit-bindgen` must be pinned to a specific `rev`.
+This will change in the future.
```

Comment:
We should really re-export these macros from the sdk...if possible? :thinking:

Reply:
That wasn't possible with one of the macros from `wit-bindgen`, we should definitely try again.